src/spikeinterface/core/sparsity.py

```diff
@@ -119,6 +125,85 @@ def unit_id_to_channel_indices(self):
                 self._unit_id_to_channel_indices[unit_id] = channel_inds
         return self._unit_id_to_channel_indices
 
+    def sparsify_waveforms(self, waveforms: np.ndarray, unit_id: str) -> np.ndarray:
```

Comment:
unit_id is not always a str can be int

Reply:
good point!

Reply:
Yes, I forgot to do this. In fact, mea culpa, I kind of new it when I did it but I ... wished that unit_ids were only string and intentionally forgot : P

I will push a small fix for this.

I would not like to do have a method with two different type of inputs, then 1/4 of the code is used to validate the input with if cases.